pkg/features/kube_features.go

```diff
@@ -169,49 +171,61 @@ const (
+	CloudControllerManager = "cloud-controller-manager"
 )
 
 func init() {
-	utilfeature.DefaultFeatureGate.Add(defaultKubernetesFeatureGates)
+	if err := utilfeature.DefaultFeatureGate.Add(defaultKubernetesFeatureGates); err != nil {
+		panic(fmt.Sprintf("failed to add features: %v", err))
+	}
 }
 
 // defaultKubernetesFeatureGates consists of all known Kubernetes-specific feature keys.
 // To add a new feature, define a key for it above and add it here. The features will be
-// available throughout Kubernetes binaries.
+// available throughout Kubernetes binaries, but will only be shown in flag help
+// string for its components.
 var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
-	ExternalTrafficLocalOnly:                    {Default: true, PreRelease: utilfeature.GA},
-	AppArmor:                                    {Default: true, PreRelease: utilfeature.Beta},
-	DynamicKubeletConfig:                        {Default: false, PreRelease: utilfeature.Alpha},
-	KubeletConfigFile:                           {Default: false, PreRelease: utilfeature.Alpha},
-	ExperimentalHostUserNamespaceDefaultingGate: {Default: false, PreRelease: utilfeature.Beta},
-	ExperimentalCriticalPodAnnotation:           {Default: false, PreRelease: utilfeature.Alpha},
-	Accelerators:                                {Default: false, PreRelease: utilfeature.Alpha},
-	DevicePlugins:                               {Default: false, PreRelease: utilfeature.Alpha},
-	TaintBasedEvictions:                         {Default: false, PreRelease: utilfeature.Alpha},
-	RotateKubeletServerCertificate:              {Default: false, PreRelease: utilfeature.Alpha},
-	RotateKubeletClientCertificate:              {Default: true, PreRelease: utilfeature.Beta},
-	PersistentLocalVolumes:                      {Default: false, PreRelease: utilfeature.Alpha},
-	LocalStorageCapacityIsolation:               {Default: false, PreRelease: utilfeature.Alpha},
-	HugePages:                                   {Default: false, PreRelease: utilfeature.Alpha},
-	DebugContainers:                             {Default: false, PreRelease: utilfeature.Alpha},
-	PodPriority:                                 {Default: false, PreRelease: utilfeature.Alpha},
-	EnableEquivalenceClassCache:                 {Default: false, PreRelease: utilfeature.Alpha},
-	TaintNodesByCondition:                       {Default: false, PreRelease: utilfeature.Alpha},
-	MountPropagation:                            {Default: false, PreRelease: utilfeature.Alpha},
-	ExpandPersistentVolumes:                     {Default: false, PreRelease: utilfeature.Alpha},
-	CPUManager:                                  {Default: false, PreRelease: utilfeature.Alpha},
-	ServiceNodeExclusion:                        {Default: false, PreRelease: utilfeature.Alpha},
+	ExternalTrafficLocalOnly:                    {Default: true, PreRelease: utilfeature.GA, Components: []string{KubeProxy, KubeAPIServer}},
```

Comment:
Can you make sure all the features are exactly associated to the components you list here ?

Reply:
Not sure how we can be certain of this, most likely all features maintainer will need to take a look ? The current changes was determined manually by functions and packages thats involved `utilfeature.DefaultFeatureGate.Enabled` check

Reply:
That's my main concern with this change... gates are intended to be a global toggle of a feature that is often cross-cutting across multiple components. I'd expect our recommendation to be to set `--feature-gates` identically for all kube components, not try to worry about which gates touch aspects of which components.

Reply:
@liggitt however setting feature-gates to be the same for all components seems a bit too heavy handed / not user-friendly, especially for cluster operator. Now the only way to be sure a feature is set is to set on all of them, which seems very wasteful ( requires cluster rolling restart even for not related components for example ).